src/state.py

```diff
@@ -0,0 +1,38 @@
+from typing import List, Dict, Optional
+
+class ChatState:
```

Comment:
I think convention is to call it State

Reply:
we could inherit the [Pydantic state](https://langchain-ai.github.io/langgraph/how-tos/state-model/)

Reply:
The State should not have any methods. It is an attribute class